pkg/syncer/metadatasyncer.go

```diff
@@ -1110,6 +1110,9 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 					log.Errorf("failed to get volume id for volume name: %q with err=%v", pv.Name, err)
 					continue
 				}
+			} else {
+				log.Warnf("Volume %q is not a valid vSphere volume for the pod %q", volume.PersistentVolumeClaim.ClaimName, pod.Name)
```

Comment:
Somehow we were not returning when pod is having an invalid volume. And somewhere below(Line:1158) in this function we are calling `UpdateVolumeMetadata()` with empty volume id. 
We can avoid this by just returning the volume is invalid instead of failing with empty volume id.
This change is not related to the original issue reported. However, I have added this change as an improvement

Reply:
I would suggest to log it at debug level as a customer could be using other CSI drivers as well.

Reply:
Done. Makes sense